apps/admin-gui/src/app/admin/pages/admin-page/admin-audit-log/admin-audit-log.component.ts

```diff
@@ -1,16 +1,49 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { TABLE_AUDIT_MESSAGES } from '@perun-web-apps/config/table-config';
+import { SelectionModel } from '@angular/cdk/collections';
+import { AuditMessage, AuditMessagesManagerService } from '@perun-web-apps/perun/openapi';
 
 @Component({
   selector: 'app-admin-audit-log',
   templateUrl: './admin-audit-log.component.html',
   styleUrls: ['./admin-audit-log.component.scss'],
 })
-export class AdminAuditLogComponent {
+export class AdminAuditLogComponent implements OnInit {
   tableId = TABLE_AUDIT_MESSAGES;
   refresh = false;
 
+  selection = new SelectionModel<AuditMessage>(true, []);
+
+  selectedEvents: string[] = [];
+  eventOptions: string[] = [];
+
+  constructor(private auditMessagesManagerService: AuditMessagesManagerService) {}
+
+  ngOnInit(): void {
+    this.auditMessagesManagerService.findAllPossibleEvents().subscribe((res) => {
+      this.eventOptions = res.sort();
+    });
+  }
+
   refreshTable(): void {
     this.refresh = !this.refresh;
   }
+
+  searchFunction = (entity: string): string => entity;
+
+  mainTextFunction = (entity: string): string => entity;
+
+  secondaryTextFunction = (_: string): string => '';
+
+  toggleEvent(event: string): void {
+    if (this.selectedEvents.includes(event)) {
+      const index = this.selectedEvents.indexOf(event);
+      this.selectedEvents.splice(index, 1);
+    } else {
+      this.selectedEvents.push(event);
+    }
+
+    // change this array to force refresh of component and calling endpoint
+    this.selectedEvents = [...this.selectedEvents];
+  }
```

Comment:
Only propagate the changes after user closes the selection. That way we only refresh the table once users select all events they need. You can take a look of how it is done for selecting member status on vo member page.